recommenders/models/deeprec/deeprec_utils.py

```diff
@@ -303,174 +302,97 @@ def load_yaml(filename):
         raise IOError("load {0} error!".format(filename))
 
 
+class HParams():
+    """Class for holding hyperparameters for DeepRec algorithms.
+    """
+    def __init__(self, hparams_dict):
+        """Create an HParams object from a dictionary of hyperparameter values.
+
+        Args:
+            hparams_dict (dict): Dictionary with the model hyperparameters.
+        """
+        for val in hparams_dict.values():
+            if not (isinstance(val, int) or isinstance(val, float) or isinstance(val, str) or isinstance(val, list)):
+                raise ValueError("Hyperparameter value {} should be integer, float, string or list.".format(val))  
+        self._values = hparams_dict
+        for hparam in hparams_dict:
+            setattr(self, hparam, hparams_dict[hparam])
+
+    def __repr__(self):
+        return "HParams object with values {}".format(self._values.__repr__())
+
+    def values(self):
+        """Return the hyperparameter values as a dictionary.
+
+        Returns:
+            dict: Dictionary with the hyperparameter values.
+        """
+        return self._values
+
+
 def create_hparams(flags):
     """Create the model hyperparameters.
 
     Args:
         flags (dict): Dictionary with the model requirements.
 
     Returns:
-        tf.contrib.training.HParams: Hyperparameter object in TF.
+        HParams: Hyperparameter object.
     """
-    return tf.contrib.training.HParams(
-        # data
-        kg_file=flags["kg_file"] if "kg_file" in flags else None,
-        user_clicks=flags["user_clicks"] if "user_clicks" in flags else None,
-        FEATURE_COUNT=flags["FEATURE_COUNT"] if "FEATURE_COUNT" in flags else None,
-        FIELD_COUNT=flags["FIELD_COUNT"] if "FIELD_COUNT" in flags else None,
-        data_format=flags["data_format"] if "data_format" in flags else None,
-        PAIR_NUM=flags["PAIR_NUM"] if "PAIR_NUM" in flags else None,
-        DNN_FIELD_NUM=flags["DNN_FIELD_NUM"] if "DNN_FIELD_NUM" in flags else None,
-        n_user=flags["n_user"] if "n_user" in flags else None,
-        n_item=flags["n_item"] if "n_item" in flags else None,
-        n_user_attr=flags["n_user_attr"] if "n_user_attr" in flags else None,
-        n_item_attr=flags["n_item_attr"] if "n_item_attr" in flags else None,
-        iterator_type=flags["iterator_type"] if "iterator_type" in flags else None,
-        SUMMARIES_DIR=flags["SUMMARIES_DIR"] if "SUMMARIES_DIR" in flags else None,
-        MODEL_DIR=flags["MODEL_DIR"] if "MODEL_DIR" in flags else None,
+    init_dict = {
```

Comment:
Good point. 
So what happened was that in TF2, `tf.contrib` has been retired. HParams was defined in `tf.contrib` but there has been no new definition in TF2 (not yet at least). So I had to define a custom class for HParams. 
Moreover, TF1 accepted `None` values for hparams but TF2 does not and raises errors (because it does stricter type checking). So I had to remove `None` values from the default dictionary. 
There are just a few references of these in the code and I have updated them e.g.
https://github.com/microsoft/recommenders/blob/7d36296bdf58eb3f4c1bf2e6fa19866baf580a84/recommenders/models/deeprec/models/base_model.py#L37
In other cases, it makes sense that there is an error if the value of an hparam is not defined in the yaml file.